pkg/bwagreement/server.go

```diff
@@ -87,6 +100,7 @@ func (s *Server) BandwidthAgreements(stream pb.Bandwidth_BandwidthAgreementsServ
 			return nil
 		case agreement := <-ch:
 			go func() {
+				defer s.locked()()
```

Comment:
Why use a goroutine when you are going to lock the whole server?

Reply:
Done. removed the go fun()